### DISCUSSION TITLE
Min/Max Grain Size

### DISCUSSION THREAD

**Discussion by janzenchoi** [neperfepx/neper#578] [Q&A] [answered]
Hello,

The distribution of my equivalent diameters are defined such that the standard deviation is larger than the mean. This introduces several problems, specifically very tiny and very large grains. I tried defining lower and upper bounds for the grain sizes, using the `interval` keyword, as shown below.

```
neper -T -n from_morpho -morpho "diameq:interval(0.0174,2.0158):lognormal(0.20766,0.27155),1-sphericity:interval(0.02316,0.57725):lognormal(0.21411,0.09016)" -dim 3 -domain "cube(1,1,1)" -statcell diameq,sphericity
```

However, the grain sizes were still out of the specified bounds. As such, I was wondering whether it is possible to define a minimum and maximum grain size with the lognormal distribution, and what that command would look like.

Thanks in advance.

**Comment by rquey** [comment, marked as answer]
`diameq:interval(0.0174,2.0158):lognormal(0.20766,0.27155)` is not proper syntax. `diameq:interval(0.0174,2.0158),diameq:lognormal(0.20766,0.27155)` works in terms of syntax, but would not be very good practice.

In your case, you can specify endpoints using `from` and `to`: `diameq:lognormal(0.20766,0.27155,from=0.0174,to=2.0158)`, as described in [Statistical Distributions](https://neper.info/doc/exprskeys.html#statistical-distributions).

**Reply by janzenchoi** [reply]
Thanks for the response, @rquey.

I took your advice, and ran the following command.

`neper -T -n from_morpho -morpho "diameq:lognormal(0.20766,0.27155,from=0.0174,to=2.0158),1-sphericity:lognormal(0.21411,0.09016)" -dim 3 -domain "cube(1,1,1)" -statcell diameq,sphericity`

However, it gave me 259 grains. Previously, when I ran the command without the bounds, the tessellation would have about 25 grains. Only a few of those grains would be outside of the bounds (i.e., `[0.0174, 2.0158]`). As such, I feel like there shouldn't be this many grains.

Do you know where I went wrong?

**Reply by rquey** [reply]
There was indeed a bug. Please use version 4.5.1-2.

Note that, in principle, when using `from` and `to`, you should preferably make sure that the resulting truncation does not change the mean value of the distribution (it may be the case with your command, I'm not sure). Otherwise, a correction has to be made and some values may go slightly out of bounds...

Here is what I now get with your command: 25 cells, min = 0.023, max = 1.088, mean = 0.217, standard deviation = 0.237.

**Reply by janzenchoi** [reply]
Updating Neper to the newest version does help. Thanks again for all the help.
